shellstub.py

```
@@ -0,0 +1,149 @@
+        exit_code: int = 0,
+        func: Callable[[Sequence[str]], int] | None = None,
+    ) -> None:
+        """Register a new stub with either a variants list or single behaviour."""
+        if variants is None:
+            variants = [
+                {
+                    "match": None,
+                    "stdout": stdout,
+                    "stderr": stderr,
+                    "exit_code": exit_code,
+                }
```

Ruff (0.11.9)

Trailing comma missing

Add trailing comma

(COM812)